### PR TITLE
Fix Conv BN folding with deadcode

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #124808

**Summary**
Fix issue: https://github.com/pytorch/pytorch/issues/124286
The TorchBenchmark includes a method called `run_n_iterations` which runs model multiple times.
https://github.com/pytorch/pytorch/blob/43f4e71daa6dce6014d30da046d28f14cf30d5a4/benchmarks/dynamo/common.py#L2272-L2276

https://github.com/pytorch/pytorch/pull/123399 enables tracing into a `UserDefinedObjectVariable` that's an instance method.    It will trace the model into FX graph multiple times within `run_n_iterations`. Then, in the Inductor, `Conv-BN folding` at the module level will fuse the same Conv-BN module multiple times in this case, which leads to accuracy failures. This PR addresses the issue by ensuring that each Conv-BN module is fused only once.

**TestPlan**
```
python -u -m pytest -s -v test/inductor/test_inductor_freezing.py -k test_folded_conv_bn_with_module_sharing
python -u -m pytest -s -v test/inductor/test_inductor_freezing.py -k test_folded_conv_functional_bn_with_module_sharing
python -u -m pytest -s -v test/inductor/test_inductor_freezing.py -k test_conv_bn_with_multi_bn_share_conv
python -u -m pytest -s -v test/inductor/test_inductor_freezing.py -k test_conv_functional_bn_with_multi_bn_share_conv
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by jansel — APPROVED** [review]

File: test/inductor/test_inductor_freezing.py [hunk 1]
```diff
@@ -375,6 +471,139 @@ def foo(mod, x):
...
+
+        def foo(mod, x):
+            return mod(x)
+
+        with torch.no_grad():
+            out_eager = foo(mod, x)
+            out_optimized_for_infernece, _ = run_and_get_code(
+                torch.compile(foo), mod, x
+            )
+        self.assertEqual(out_optimized_for_infernece, out_eager, atol=1e-2, rtol=1e-2)
```
Comment (jgong5) [inline test/inductor/test_inductor_freezing.py:605]: Shall we also check if BN still exists in the graph?

File: test/inductor/test_inductor_freezing.py [hunk 2]
```diff
@@ -375,6 +471,139 @@ def foo(mod, x):
...
+        def foo(mod, x):
+            return mod(x)
+
+        with torch.no_grad():
+            out_eager = foo(mod, x)
+            out_optimized_for_infernece, _ = run_and_get_code(
+                torch.compile(foo), mod, x
+            )
+
+        self.assertEqual(out_optimized_for_infernece, out_eager, atol=1e-2, rtol=1e-2)
```
Comment (jgong5) [inline test/inductor/test_inductor_freezing.py:570]: Shall we also check if BN still exists in the graph?

**Review by jgong5 — APPROVED** [review]

File: test/inductor/test_inductor_freezing.py (hunk 1 above)
Comment (leslie-fang-intel) [reply test/inductor/test_inductor_freezing.py:590]: Actually, the BN has been decomposed in https://github.com/pytorch/pytorch/blob/57790fd088af40d9ab44c66def34db23f9d2879f/torch/_decomp/decompositions.py#L1704. Remove related checkers.

File: test/inductor/test_inductor_freezing.py (hunk 2 above)
Comment (leslie-fang-intel) [reply test/inductor/test_inductor_freezing.py:555]: Same as above.
